galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/auth/IAuthStore.java

```diff
@@ -44,5 +45,41 @@ public interface IAuthStore {
      */
     void deleteToken(String tokenId) throws AuthStoreException;
 
+    /**
+     * Returns a list of all the users using the system.
+     *
+     * @return a list of all users stored in the users store.
+     * @throws AuthStoreException if there is an issue accessing the users store.
+     */
+    List<IUser> getAllUsers() throws AuthStoreException;
+
+    /**
+     * Stores a new user record in the users store's database.
+     *
+     * @param loginId    the loginId of the user trying to access Galasa API
+     * @throws AuthStoreException if there is an issue accessing the users store.
+     */
+    void createUser(String loginId, String clientName) throws AuthStoreException;
+
+    /**
+     * Retrieves a user record in the users store's database.
+     *
+     * @param loginId    the loginId of the user trying to access Galasa API
+     * @throws AuthStoreException if there is an issue accessing the users store.
+     */
+    IUser getUserByLoginId(String loginId) throws AuthStoreException;
+
+    /**
+     * Retrieves a user record in the users store's database.
+     *
```

Comment:
Should this be "Updates a user record..."? Same applies for changes in IAuthStoreService